chore: update-naming

ryanbas21 opened the pull request:
# JIRA Ticket

N/A

## Description

General infrastruture changes for the new repo.

- Setup changesets
- Ignored all packages by default for now
- Using pnpm catalog for shared dependencies
- Added a create-package generator
- Added documentation for the create-pacakge-generator
- Added further documentation to the contributing_docs folder

ryanbas21 committed:
chore: update-naming

ryanbas21 committed:
feat(nx-cloud): setup nx cloud workspace 
    
This commit sets up Nx Cloud for your Nx workspace, enabling distributed caching and the Nx Cloud GitHub integration for fast CI and improved developer experience.

You can access your Nx Cloud workspace by going to 
https://cloud.nx.app/orgs/6195684a2f082e00055e3c09/workspaces/6762f3018aed9c5fad7009bb

**Note:** This commit attempts to maintain formatting of the nx.json file, however you may need to correct formatting by running an nx format command and committing the changes.

ryanbas21 committed:
Merge pull request #2 from ForgeRock/feat/nx-cloud/setup

feat(nx-cloud): set up nx workspace

ryanbas21 committed:
chore: formatting

ryanbas21 committed:
chore: add-slack-message-on-publish

ryanbas21 committed:
chore: update-e2e-tasks

ryanbas21 committed:
chore: eslint-inferred-plugin

ryanbas21 committed:
chore: run-scripts-as-inferred

ryanbas21 committed:
chore: vite-inferred

ryanbas21 committed:
chore: remove-setup-file-from-create-package

ryanbas21 committed:
test: test-generator-execution

ryanbas21 committed:
chore: rename-e2e-ci

ryanbas21 committed:
chore: use-pnpm-catalog

ryanbas21 committed:
chore: rename-to-forgerock

ryanbas21 committed:
chore: remove-legacy-cache

cerebrl commented on tools/create-package/README.md:
```diff
@@ -1,7 +1,11 @@
 ## Building
 
-Run `nx build create-package` to build the library.
+Run `pnpm nx build create-package` to build the library.
```
Sorry if this is a silly question, but what's the difference between `pnpm create-package` and `pnpm nx build create-package`?

ryanbas21 replied:
They are identical. The main difference is that `pnpm create-package` is an alias for `pnpm nx build create-package`. I'm fine referencing the `alias`, however, when looking at the `documentation` for the actual package that is being run, it _may_ be better to reference the actual command.

Open to a different thought though

cerebrl replied:
Okay, the reason I brought this up was if they were the same, let's pick one, rather than use both throughout our docs.

ryanbas21 replied:
Lets add this to a new PR for now

cerebrl reviewed (commented):
Just a cursory scan of the changes. I'll go through it again a bit later.

ryanbas21 commented on README.md:
```diff
@@ -1,127 +1,13 @@
-
-&copy; Copyright 2020 ForgeRock AS. All Rights Reserved.
-
-[forgerock-logo]: https://www.forgerock.com/themes/custom/forgerock/images/fr-logo-horz-color.svg 'ForgeRock Logo'
```
I think we should because its copyright of a company that doesn't exist. if we want to copywright it, should it not be for the new `Ping` name?

ryanbas21 commented on contributing_docs/TOC.md:
```diff
@@ -0,0 +1,42 @@
+  - [Configuration Options](./creating-a-package.md#options)
+    - [Package Naming](./creating-a-package.md#options)
+    - [Scoping](./creating-a-package.md#optional-prompts)
+    - [Side Effects](./creating-a-package.md#optional-prompts)
```
side effects for package.json.

ryanbas21 commented on README.md:
```diff
@@ -1,127 +1,13 @@
+</p>
 
-The ForgeRock JavaScript SDK enables you to quickly integrate the [ForgeRock Identity Platform](https://www.forgerock.com/digital-identity-and-access-management-platform) into your client-side JavaScript apps.
+The Ping SDK for Javascript is designed for creating web applications that seamlessly integrate with the PingOne platform.
+It offers a range of APIs for user authentication, user device management, and accessing resources secured by PingOne.
```
Fair enough i can make this change

ryanbas21 committed:
chore: add-license

added license file to each package, added to the generation of a package
using create-package, and also added it to the bottom of the readme.

ryanbas21 committed:
test: add-test-for-license

added a test for the license generation

ryanbas21 commented on .changeset/config.json:
```diff
@@ -3,16 +3,17 @@
   "linked": [],
   "access": "public",
-  "baseBranch": "master",
+  "baseBranch": "main",
```
using only main branch now (unless we otherwise state differently).

ryanbas21 commented on .changeset/config.json:
```diff
@@ -3,16 +3,17 @@
+  "baseBranch": "main",
   "updateInternalDependencies": "patch",
   "ignore": [
+    "@forgerock/*",
```
ignoring everything (my next pr adds a changeset)

ryanbas21 commented on .github/workflows/publish.yml:
```diff
@@ -76,21 +75,14 @@ jobs:
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          NPM_CONFIG_PROVENANCE: true
+      - name: Send GitHub Action data to a Slack workflow
```
adding a slack hook to send published packages as json to a slack message&channel

ryanbas21 commented on LICENSE:
standard LICENSE file with updated year.

ryanbas21 commented on contributing_docs/README.md:
```diff
@@ -0,0 +1,42 @@
+# 📚 Project Documentation
```
renamed this to `README.md` so github displays it nicely.

cerebrl reviewed (approved):
This looks good. I left one comment about using one command OR the other, if they do the same thing, but it's not blocking.
